### PR TITLE
Remove current interactive mode

### PR DESCRIPTION
**Pull request by coolo**
I'm convinced it will never fully work, so first step to replace it with
something that scales

### CODE REVIEW

**Comment by coolo** [comment]
See https://hackweek.suse.com/16/projects/openqa-ide

**Comment by okurz** [comment]
well, a valid approach for developing but please don't merge this. The current interactive mode barely works but still well enough to provide a benefit over doing everything manually by restarting jobs. Probably you know by now that I am not a friend of "complete rewrites" anyway ;)

**Comment by coolo** [comment]
Problably you know by now that I know better and you shouldn't argue with me about such matters :)

**Comment by okurz** [comment]
I know you well enough that I assume you could actually merge the removal, yes :)
If you actually do this? Well, you have preserved that mistery cloud well enough that I simply don't know ;)
I also know that you listen to people ... you just are good at pretending in not doing so

File: lib/OpenQA/Scheduler.pm [hunk 1]
```diff
@@ -48,7 +48,7 @@ use constant SCHEDULE_TICK_MS => $ENV{OPENQA_SCHEDULER_SCHEDULE_TICK_MS} // 8000
 use constant CONGESTION_CONTROL => $ENV{OPENQA_SCHEDULER_CONGESTION_CONTROL} // 1;
 
 # Wakes up the scheduler on request
-use constant WAKEUP_ON_REQUEST => $ENV{OPENQA_SCHEDULER_WAKEUP_ON_REQUEST} // 0;
+use constant WAKEUP_ON_REQUEST => $ENV{OPENQA_SCHEDULER_WAKEUP_ON_REQUEST} // 1;
```
[the remark this answers is not in the document]
Comment (Martchus) [reply lib/OpenQA/Scheduler.pm]: At least for my local use setting this variable makes sense (I suppose). But yes, maybe better in a separate commit.

[the remark this answers is not in the document]
Comment (mudler) [reply lib/OpenQA/Scheduler.pm]: Even tho, a separate commit is useful to track the change

[the remark this answers is not in the document]
Comment (coolo) [reply lib/OpenQA/Scheduler.pm]: I didn't even mean to commit it :)

**Comment by mudler** [comment]
@okurz i don't see benefit of having code parts that are buggy, makes the system unstable and difficult to maintain. 

+1 from my side with the mindset of replacing it with a proper mechanism

**Comment by foursixnine** [comment]
+1 from me too, as long as a version is tagged and published. We can pull out the scythe!

**Comment by coolo** [comment]
I removed the sneaky parts now - it's only what is promised in the commit log

**Comment by coolo** [comment]
no idea how my changes affect 12-needle-edit. @Martchus any idea?

**Comment by okurz** [comment]
@mudler:

> @okurz i don't see benefit of having code parts that are buggy, makes the system unstable and difficult to maintain.

I hope you are kidding with this.

I do not see you on
https://github.com/os-autoinst/os-autoinst-needles-opensuse/graphs/contributors
so I assume you don't really have good real-life experience in test maintaining regarding needles. Ask around in your office. If you find *no* openQA test developer, e.g. in QA or QAM, that mentions that the interactive mode is more useful for them than not having it all then you can surely remove it.

**Comment by mudler** [comment]
@okurz i don't play games ;) it's just broken

**Comment by mudler** [comment]
And, JFIY - github as metric sucks

**Comment by aaannz** [comment]
I agree with @okurz. Even though current "interactive" mode is not really interactive, it is tremendous time saver when we are talking about more then one needle update. Also I don't recall any issues using it last time I needed to.

**Comment by foursixnine** [comment]
@aaannz @okurz: Just to be clear: Is not that it will be removed and test developers will be left without interactive mode forever. And to avoid that, it's why I'm talking about tagging a version before merging this change and moving forward, so that people is not that affected while the new interactive mode is being developed.

**Comment by Martchus** [comment]
@coolo About `12-needle-edit.t`: A reason your changes could have triggered it would be broken JavaScript code. But then likely more would fail. So I doubt that your changes triggered the failing `12-needle-edit.t`. I assume you can't reproduce locally?

@foursixnine We should not only tag a stable version, but also ensure people are using packages/repos containing only that tagged version. This is important because downgrading afterwards might be difficult (eg. database migration already happened).

**Comment by okurz** [comment]
@foursixnine no need to have a *tag* when you have a *branch* already, it's called `coolo:enhanced_live` and lives in https://github.com/coolo/openQA/tree/enhanced_live

**Comment by foursixnine** [comment]
@Martchus that was my point precisely :) but good that is noted.

**Comment by Martchus** [comment]
@okurz We want to merge this to master so openQA developers (not users) can develop new features based on it. This way we can avoid conflicts. Also, the new pieces required for the next interactive mode will be merged step by step so we can continuously test them during development.

Collecting everything on `enhanced_live` and not merging it until the next interactive mode is complete would lead this becoming a monster PR. Hard to review, hard to rebase, hard to test by other developers.

BTW: Letting the users pull directly from master is a bad idea anyways. Almost no projects do it that way. Most use version tags as @foursixnine suggests.

**Comment by dzedro** [comment]
Honestly, interactive mode doesn't work properly for me when I use remote workers, @mudler  mentioned there is some hack for it ... but I don't care, I have no time to dig around to make it work. I already don't miss it. When this will bring something better, [toi toi toi](https://www.youtube.com/watch?v=GbKAaSf6e10&feature=youtu.be&t=7s) :hammer:

**Comment by coolo** [comment]
https://github.com/os-autoinst/os-autoinst/pull/958 is the reason I need to remove this from the webui - to develop isotovideo freely. And doing this properly will take time. And we will develop this in master

And you can find *me* in https://github.com/os-autoinst/os-autoinst-needles-opensuse/graphs/contributors, no? I do know the value of the interactive mode - when it used to work

**Comment by okurz** [comment]
@coolo @Martchus I trust that you would be able to implement a proper interactive mode in a branch within a short time which would simply replace the old interactive mode within the same PR. Ok, two actually, one for os-autoinst and one for openQA. This is why I think you would need to merge the deletion first.

@Martchus And arguing that the PR would become a monster branch means to me something like: "Any new interactive mode would not be finished fast". So seems like you are contradicting yourself.

Ok, I raised my concerns, you can go forward: Surprise me with whatever outcome ;)

**Comment by coolo** [comment]
I don't think anyone but you said it would be done quickly. Noone said that - @Martchus's words were continously and 'step by step'. Deleting the often not working and unmaintained interactive mode is just the first step.

File: assets/javascripts/needleeditor.js [hunk 2]
```diff
@@ -335,7 +335,7 @@ function reactToSaveNeedle(data) {
   }
   if (data.info) {
     var info = data.info;
-    if (data.interactive_job) {
+    if (data.interactive_job) { # FIXME: never set
```
Comment (coolo) [inline assets/javascripts/needleeditor.js]: @Martchus - see this? :)

**Comment by coolo** [comment]
Just checked with top openSUSE needlers: they don't use interactive mode.

**Comment by asdil12** [comment]
+1 from my side

**Pull request merged**